component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/core/dao/impl/CacheBackedJWTConfigurationDAOImpl.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.core.dao.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.core.dao.JWTAuthenticationConfigurationDAO;
+import org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.core.dao.cache.JWTConfigCache;
+import org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.core.dao.cache.JWTConfigCacheEntry;
+import org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.core.dao.cache.JWTConfigCacheKey;
+import org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.core.exception.JWTClientAuthenticatorServiceServerException;
+import org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.core.model.JWTClientAuthenticatorConfig;
+
+/**
+ * Cached DAO layer for JWT Authenticator Configurations.
+ * All the DAO access should happen through this layer to ensure single point of caching.
+ */
+public class CacheBackedJWTConfigurationDAOImpl implements JWTAuthenticationConfigurationDAO {
+
+    private static final Log log = LogFactory.getLog(CacheBackedJWTConfigurationDAOImpl.class);
+
+    private final JWTAuthenticationConfigurationDAO privateKeyJWTAuthenticationConfigurationDAO;
+
+    public CacheBackedJWTConfigurationDAOImpl(JWTAuthenticationConfigurationDAO
+                                                      privateKeyJWTAuthenticationConfigurationDAO) {
+
+        this.privateKeyJWTAuthenticationConfigurationDAO = privateKeyJWTAuthenticationConfigurationDAO;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getPriority() {
+
+        return 5;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public JWTClientAuthenticatorConfig getPrivateKeyJWTClientAuthenticationConfigurationByTenantDomain(
+            String tenantDomain)
+            throws JWTClientAuthenticatorServiceServerException {
+
+        JWTClientAuthenticatorConfig cachedResult = getJWTConfigurationFromCache(tenantDomain);
+        if (cachedResult != null) {
+            return cachedResult;
+        }
+
```

Comment:
Let's add an info log here mentioning that the data will be retrieved from the db as cache is null.

Reply:
Fixed with https://github.com/wso2-extensions/identity-oauth-addons/pull/100/commits/d07baabaa4558de269ce04e390a0674e33191160